src/Common/Services.fs

```diff
@@ -109,6 +109,36 @@ module Result =
         | Ok _ -> true
         | Error _ -> false
 
+    /// Zips a pair of Oks into an Ok of pairs
+    /// In case at least one of the argument contains the Error case,
+    /// simply concatenates the errors.
+    let combine result1 result2 =
```

Comment:
Why are `combine` and `contract` moved to Common.Service?

Reply:
Not only needed in typechecking but also in annotation checking and later on in namechecking.